compiler/generator/jax/jax_instructions.hh

```diff
@@ -0,0 +1,803 @@
+    
+    JAXInitFieldsVisitor(std::ostream* out, int tab = 0) : fOut(out), fTab(tab) {}
+    
+    virtual void visit(DeclareVarInst* inst)
+    {
+        ArrayTyped* array_type = dynamic_cast<ArrayTyped*>(inst->fType);
+        if (array_type) {
+            tab(fTab, *fOut);
+            inst->fAddress->accept(this);
+            *fOut << " = ";
+            if (inst->fValue) {
+                inst->fValue->accept(this);
+            } else {
+                ZeroInitializer(fOut, inst->fType);
+            }
+        }
+    }
+    
+    virtual void visit(NamedAddress* named)
+    {
+        // kStaticStruct are actually merged in the main DSP
+        if (named->getAccess() & Address::kStruct || named->getAccess() & Address::kStaticStruct) {
+            *fOut << "state[\"";
+        }
+        *fOut << named->fName;
+        if (named->getAccess() & Address::kStruct || named->getAccess() & Address::kStaticStruct) {
+            *fOut << "\"]";
+        }
+    }
+    
+    static void ZeroInitializer(std::ostream* fOut, Typed* typed)
+    {
+        ArrayTyped* array_type = dynamic_cast<ArrayTyped*>(typed);
+        faustassert(array_type);
+        if (isIntPtrType(typed->getType())) {
+            *fOut << "np.zeros((" << array_type->fSize << ",), dtype=np.int32)";
+        } else if (isFloatType(typed->getType())) {
+            *fOut << "np.zeros((" << array_type->fSize << ",), dtype=np.float32)";
+        } else {
+            *fOut << "np.zeros((" << array_type->fSize << ",), dtype=np.float64)";
+        }
+    }
+    
+    // Needed for waveforms
+    
+    virtual void visit(Int32ArrayNumInst* inst)
+    {
+        *fOut << "np.array(";
+        char sep = '[';
+        for (size_t i = 0; i < inst->fNumTable.size(); i++) {
+            *fOut << sep << inst->fNumTable[i];
+            sep = ',';
+        }
+        *fOut << "], dtype=np.int32)";
+    }
+    
+    virtual void visit(FloatArrayNumInst* inst)
+    {
+        *fOut << "np.array(";
+        char sep = '[';
+        for (size_t i = 0; i < inst->fNumTable.size(); i++) {
+            *fOut << sep << checkFloat(inst->fNumTable[i]);
+            sep = ',';
+        }
+        *fOut << "], dtype=np.float32)";
+    }
+    
+    virtual void visit(DoubleArrayNumInst* inst)
+    {
+        *fOut << "np.array(";
+        char sep = '[';
+        for (size_t i = 0; i < inst->fNumTable.size(); i++) {
+            *fOut << sep << checkDouble(inst->fNumTable[i]);
+            sep = ',';
+        }
+        *fOut << "], dtype=np.float64)";
+    }
+    
+};
+
+class JAXInstVisitor : public TextInstVisitor {
+   private:
+     
+    /*
+     Global functions names table as a static variable in the visitor
+     so that each function prototype is generated as most once in the module.
+     */
+    static map<string, bool> gFunctionSymbolTable;
+
+    // Polymorphic math functions
+    map<string, string> gPolyMathLibTable;
+    
+    bool fMutateFun;
+    bool is_storing_lhs = false;
```

Comment:
Can you follow the naming/formatting code convention we have for class field ? So  `is_storing_lhs` would become `fIsStoringLhs` and same for other ones.